core/src/data_sources/folder.rs

```diff
@@ -1,8 +1,84 @@
 use serde::{Deserialize, Serialize};
 
-#[derive(Debug, Clone, Serialize, PartialEq, Deserialize)]
+use crate::project::Project;
+
+use super::node::{Node, NodeType, SimpleNode};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct Folder {
-    pub data_source_id: String,
-    pub folder_id: String,
-    pub created: u64,
+    project: Project,
+    data_source_id: String,
+    folder_id: String,
+    created: u64,
+    timestamp: u64,
+    title: String,
+    mime_type: String,
```

Comment:
Do we expect to have different mime types for folders ?

Reply:
Should we have a constant somewhere for this ?

Reply:
yes, definitely - we don't expect anything else than `application/vnd.dust.folder`